docs/source/providers/index.md

```diff
@@ -0,0 +1,45 @@
+# Providers Overview
+
+The goal of Llama Stack is to build an ecosystem where users can easily swap out different implementations for the same API. Examples for these include:
+- LLM inference providers (e.g., Fireworks, Together, AWS Bedrock, Groq, Cerebras, SambaNova, etc.),
+- Vector databases (e.g., ChromaDB, Weaviate, Qdrant, FAISS, PGVector, etc.),
+- Safety providers (e.g., Meta's Llama Guard, AWS Bedrock Guardrails, etc.)
+
+Providers come in two flavors:
+- **Remote**: the provider runs as a separate service external to the Llama Stack codebase. Llama Stack contains a small amount of adapter code.
+- **Inline**: the provider is fully specified and implemented within the Llama Stack codebase. It may be a simple wrapper around an existing library, or a full fledged implementation within Llama Stack.
+
+Importantly, Llama Stack always strives to provide at least one fully "local" provider for each API so you can iterate on a fully featured environment locally.
+
+## Agents
+
+## DatasetIO
+
+## Eval
+
+## Inference
+
+## iOS
+
+## Post Training
+
+## Safety
+
+## Scoring
+
+## Telemetry
+
+## Tool Runtime
+
+## Vector DBs
+
+```{toctree}
+:maxdepth: 1
+
+vector_db/chromadb
```

Comment:
nit: this should be `vector_io` probably... it's a bit confusing but "vector_db" is the resource and "vector_io" is the provider operating on that resource. just like "models" are the resource and "inference" is the operational API for which there is a provider.

Reply:
Updated.